contracts/mechanics/garden-of-forking-paths/GardenOfForkingPaths.sol

```diff
@@ -1078,44 +1085,85 @@ contract GOFPFacet is
-                } else if (
+            if (stageReward.rewardAddress != address(0)) {
+                if (
+                    lastStage == 0 ||
                     gs.pathChoices[sessionId][tokenIds[i]][lastStage] ==
                     lastStageCorrectPath
                 ) {
-                    stageRewardAmount += stageReward.rewardAmount;
+                    if (stageReward.inventoryAddress == address(0)) {
+                        stageRewardAmount += stageReward.rewardAmount;
+                    } else {
+                        _distributeReward(stageReward, tokenIds[i]);
+                    }
                 }
             }
 
             // Distribute path reward.
-            Reward storage pathReward = gs.sessionPathReward[sessionId][
+            Reward memory pathReward = gs.sessionPathReward[sessionId][
                 currentStage
             ][paths[i]];
-            if (pathReward.terminusAddress != address(0)) {
+            _distributeReward(pathReward, tokenIds[i]);
+        }
+
+        if (stageReward.inventoryAddress == address(0)) {
+            stageReward.rewardAmount = stageRewardAmount;
+            _distributeReward(stageReward, 0);
+        }
+    }
+
+    function _distributeReward(Reward memory reward, uint256 tokenId) internal {
+        if (reward.rewardAddress != address(0)) {
+            if (reward.rewardType == TERMINUS_MINTABLE_TYPE) {
                 TerminusFacet rewardTerminus = TerminusFacet(
-                    pathReward.terminusAddress
-                );
-                rewardTerminus.mint(
-                    msg.sender,
-                    pathReward.terminusPoolId,
-                    pathReward.rewardAmount,
-                    ""
+                    reward.rewardAddress
                 );
+                if (reward.inventoryAddress != address(0)) {
+                    rewardTerminus.mint(
+                        address(this),
+                        reward.rewardTokenID,
+                        reward.rewardAmount,
+                        ""
+                    );
+                    _equipInventoryReward(reward, tokenId);
+                } else {
+                    rewardTerminus.mint(
+                        msg.sender,
+                        reward.rewardTokenID,
+                        reward.rewardAmount,
+                        ""
+                    );
+                }
+            } else {
+                revert("Non-terminus rewards are not yet implemented");
             }
         }
+    }
 
-        if (stageReward.terminusAddress != address(0)) {
-            TerminusFacet rewardTerminus = TerminusFacet(
-                stageReward.terminusAddress
-            );
-            rewardTerminus.mint(
-                msg.sender,
-                stageReward.terminusPoolId,
-                stageRewardAmount,
-                ""
-            );
+    function _equipInventoryReward(
+        Reward memory reward,
+        uint256 tokenId
```

Comment:
@kellan-simiotics : I believe we may want to add an argument `boolean force` which sets inventory slots to impersistent to equip the rewards and then atomically makes them persistent again.

This can only be used if the GOFP contract is an Inventory admin, but it will be useful for things like hit points, experience points, etc.

Reply:
This can be a separate discussion and a separate PR if we decide to do it. This contract is becoming quite complex, so don't want to make too many changes in a single PR.

Reply:
Disregard, you are already doing the correct thing here.